pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -43,13 +43,24 @@ import (
 	"k8s.io/kubernetes/pkg/scheduler/util"
 )
 
-var (
+const (
+	// If the pod stays in unschedulableQ longer than the unschedulableQTimeInterval,
+	// the pod will be moved from unschedulableQ to activeQ.
+	unschedulableQTimeInterval = 60 * time.Second
+
 	queueClosed = "scheduling queue is closed"
 )
 
-// If the pod stays in unschedulableQ longer than the unschedulableQTimeInterval,
-// the pod will be moved from unschedulableQ to activeQ.
-const unschedulableQTimeInterval = 60 * time.Second
+const (
+	// DefaultPodInitialBackoffDuration is the default value for the initial backoff duration
+	// for unschedulable pods. To change the default podInitialBackoffDurationSeconds used by the
+	// scheduler, update the ComponentConfig value in defaults.go
+	DefaultPodInitialBackoffDuration time.Duration = 1 * time.Second
+	// DefaultPodMaxBackoffDuration is the default value for the max backoff duration
+	// for unschedulable pods. To change the default podMaxBackoffDurationSeconds used by the
+	// scheduler, update the ComponentConfig value in defaults.go
+	DefaultPodMaxBackoffDuration time.Duration = 10 * time.Second
```

Comment:
ok, it turns out that we can define constants in kubernetes/pkg/scheduler/apis/config/types.go, and as long as we don't add those constants to staging/src/k8s.io/kube-scheduler/config/v1alpha1/types.go, then they are not part of the API. What do you think about doing that?

Reply:
> ok, it turns out that we can define constants in kubernetes/pkg/scheduler/apis/config/types.go, and as long as we don't add those constants to staging/src/k8s.io/kube-scheduler/config/v1alpha1/types.go, then they are not part of the API. What do you think about doing that?

Do we have precedence on such a thing? I'm ok with it if there was one.

Reply:
It is up to us, no precedence is required here. k8s don't guarantee any level of compatibility with any source we produce that is not published for consumption as a go library (we only guarantee that for things we publish under the ```staging``` directory), we have strong guarantees only for REST APIs, config file yaml/json and binary flags. That being said, I will leave it up to you to decide.

Reply:
After some digging, I prefer to keep it **as it is**. Even if we define the constants in `pkg/scheduler/apis/config/types.go`, we still need to use hard code value here https://github.com/kubernetes/kubernetes/pull/81263/files#diff-9d72b6493a5482d8a2829af1390c5733R106-R115 since the package shouldn't depend on other k8s.io/kubernetes code.

https://github.com/kubernetes/kubernetes/blob/1e761470ad8fa06b50bce99e483f556023b99524/pkg/scheduler/apis/config/v1alpha1/defaults.go#L23-L31

Reply:
sg, please squash.

Reply:
> sg, please squash.

done, please take a look

Reply:
@draveness Regrading  https://github.com/kubernetes/kubernetes/pull/81263/files#r325523994. Are you saying `pkg/scheduler/apis/config/v1alpha1` should not import `pkg/scheduler/apis/config`?

Reply:
> @draveness Regrading https://github.com/kubernetes/kubernetes/pull/81263/files#r325523994. Are you saying `pkg/scheduler/apis/config/v1alpha1` should not import `pkg/scheduler/apis/config`?

Yes, it documents in the `pkg/scheduler/apis/config/v1alpha1/defaults.go` where I listed above